lib/ansible/modules/windows/win_toast.ps1

```diff
@@ -0,0 +1,101 @@
+#!powershell
+# This file is part of Ansible
+#
+# Copyright 2017, Jon Hawkesworth (@jhawkesworth) <figs@unity.demon.co.uk>
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$ErrorActionPreference = "Stop"
+
+# version check
+$osversion = [Environment]::OSVersion
+$lowest_version = 10
+if ($osversion.Version.Major -lt $lowest_version ) {
+   Fail-Json $result "Sorry, this version of windows, $osversion, does not support Toast notifications.  Toast notifications are available from version $lowest_version" 
+}
+
+$stopwatch = [system.diagnostics.stopwatch]::startNew()
+$now = [DateTime]::Now
+$default_title = "Notification: " + $now.ToShortTimeString()
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+# $expire_at = Get-AnsibleParam -obj $params -name "msg" -type "str"
+$expire_secs = Get-AnsibleParam -obj $params -name "expire_secs" -type "int" -default 45
```

Comment:
I am not fond of parameter options with the unit in the parameter name. We have had that discussion before, and actually decided to raise this to the core team. But usually the unit is part of the parameter description.

Besides, other parameters are using `_sec`, this is the first time `_secs` is proposed.

Reply:
I've actually changed to 'expire_seconds' (same I as used in win_msg) so it won't be _secs (which I think is not good - whole words are always clearer).

Reply:
Noted and updated the Core agenda ticket.